packages/app/lib/internal/registry/app.js

```diff
@@ -175,6 +177,16 @@ export function initializeApp(options = {}, configOrName) {
     .then(() => {
       app._initialized = true;
       return app;
+    })
+    .catch(e => {
+      // we need to clean the app entry from registry as the app does not actually exist
+      // There are still possible side effects from `onAppCreateFn` to consider but as existing
+      // code may rely on that function running prior to native create, re-ordering it is a semantic change
+      // and will be avoided
+      delete APP_REGISTRY[name];
```

Comment:
@Salakar it was even worse than your original commit where I committed copy-pasta-cide
I was trying to splice the entry out of object, and of course splice is for arrays.
Anyway, the root cause is one I/we know the answer to: actually test the thing

So now it has an e2e test which was illuminating by itself, because firebase-android-sdk simply doesn't care about bad init values while iOS throws. Who knew? But it's all exercised now with solid assertions :+1: